src/Exception/TopicNotSet.php

```diff
@@ -0,0 +1,14 @@
+<?php
+declare(strict_types=1);
+
+namespace Kafka\Exception;
+
+use Kafka\Exception;
+
+final class TopicNotSet extends Exception
```

Comment:
THB I don't feel the need for this exception, it's essentially related to an invalid record in the record set. We could combine the three exceptions you created and have something like (names can still be improved):

```php
final class InvalidRecordInSet extends Exception
{
    public static function missingTopic(): self
    {}

    public static function missingValue(): self
    {}

    public static function nonexististingTopic(string $topic): self
    {}
}
```